Core/src/org/sleuthkit/autopsy/coreutils/AppDBParserHelper.java

```diff
@@ -0,0 +1,485 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.coreutils;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.logging.Level;
+import org.apache.commons.lang3.StringUtils;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.AccountFileInstance;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.Blackboard;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.Relationship;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskDataException;
+
+
+/**
+ * A helper class to support modules that parse SQLite databases from mobile 
+ * apps and create artifacts.
+ */
+public final class AppDBParserHelper  {
```

Comment:
Can this be renamed to BlackboardUtility or something?

Reply:
I'm happy to rename it to anything we all agree on.  I'm a little loathe to call it a "Utility" because it has too much state information - a "Utility" class in my mind is all static methods without any state.

How about BlackboardHelper?


@bcarrier  had also alluded to the possibility of moving this into SleuthKit instead.  Although currently I started using some util methods from CoreUtils (NetworkUtils.extractDomain() to extract domain from a URL).  If we do want to move this to Sleuthkit then I should refrain from using any Autopsy methods....